service/storage_service.cc

```diff
@@ -747,7 +748,9 @@ future<> storage_service::sstable_cleanup_fiber(raft::server& server, sharded<se
                     co_return co_await sp.await_pending_writes();
                 });
                 auto& compaction_module = _db.local().get_compaction_manager().get_task_manager_module();
-                auto task = co_await compaction_module.make_and_start_task<cleanup_keyspace_compaction_task_impl>({}, ks_name, _db, table_infos);
+                // we flush all tables before cleanup the keyspaces individually, so skip the flush-tables step here
+                auto task = co_await compaction_module.make_and_start_task<cleanup_keyspace_compaction_task_impl>(
+                    {}, ks_name, _db, table_infos, flush_mode::all_tables);
```

Comment:
But if we flush all tables below, before invoking `do_cleanup_ks`, then here we should pass `flush_mode::skip`

Reply:
okay. the reason i used `flush_mode::all_tables` was that the caller perform the call on behalf of the individual task to flush *all* tables, so thought `flush_mode::all_tables` might be better. but yeah, `skip` is what the task should do. so changed as proposed.